src/cli/self_update/windows.rs

```diff
@@ -859,30 +842,28 @@ pub struct RegistryValueId {
 
 #[cfg(any(test, feature = "test"))]
 impl RegistryValueId {
-    pub fn get_value<T: FromRegValue>(&self) -> io::Result<Option<T>> {
-        self.get()?.map(|v| T::from_reg_value(&v)).transpose()
+    pub fn get_value(&self) -> Result<Option<Value>> {
+        self.get()
     }
 
-    fn get(&self) -> io::Result<Option<RegValue>> {
-        let sub_key = RegKey::predef(HKEY_CURRENT_USER)
-            .open_subkey_with_flags(self.sub_key, KEY_READ | KEY_WRITE)?;
-        match sub_key.get_raw_value(self.value_name) {
+    fn get(&self) -> Result<Option<Value>> {
+        let sub_key = CURRENT_USER.create(self.sub_key)?;
+        match sub_key.get_value(self.value_name) {
             Ok(val) => Ok(Some(val)),
-            Err(ref e) if e.kind() == io::ErrorKind::NotFound => Ok(None),
-            Err(e) => Err(e),
+            Err(e) if e.code() == HRESULT::from_win32(ERROR_FILE_NOT_FOUND) => Ok(None),
+            Err(e) => Err(e.into()),
         }
     }
 
-    pub fn set_value(&self, new: Option<impl ToRegValue>) -> io::Result<()> {
-        self.set(new.map(|s| s.to_reg_value()).as_ref())
+    pub fn set_value(&self, new: Option<(Type, &[u8])>) -> Result<()> {
```

Comment:
It would probably make sense to remove `(get|set)_value()` entirely in a separate commit now that they are not different from `get()` and `set()` respectively.

Reply:
I think I'd better open a seperate PR once this PR is merged.😵

Reply:
> I think I'd better open a seperate PR once this PR is merged.😵

@InfyniteHeap That'd be nice as well.